### ISSUE TITLE
Screen Reader announcing string numbers as finger numbers

### ISSUE THREAD

**Issue by jmwmusic** [musescore/MuseScore#27954]
### Issue type

Accessibility issue (e.g. for keyboard-only or screen reader users)

### Description with steps to reproduce

A circled string number added to the score from the fingering palette is incorrectly announced on the score as a finger number by the screen reader.

STEPS:
1.  click on a note and from the fingering palette choose a circled string number.  NOTE:  If you arrow over this palette these are clearly announced as string numbers not finger numbers.
2.  On the score arrow across the note the string number was added to and you will hear the screen reader announce it as a finger number, it should say string number.

This is causing an issue where string numbers are being used by people creating scores instead of a circled finger number (because there is no circled finger number) and the output in Braille is very different for string numbers than for finger numbers with a circle.  The output in Braille when using SMB is correct for these as string numbers, so the xml sees them as a string number not a circled finger number.

It would be ideal if they were in the guitar palette therefore and not in the fingering palette.

### Supporting files, videos and screenshots

N/A

### What is the latest version of MuseScore Studio where this issue is present?

4.5.2

### Regression

I was unable to check

### Operating system

Windows 11

### Additional context

Tested with NVDA screen reader.

### Checklist

- [x] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [x] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [x] I have attached all requested files and information to this report
- [x] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by shoogle** [comment]
On a techical level, string numbers in MuseScore are fingering elements that happen to use the "String number" text style.

Text styles for fingering elements include:

1. "Fingerings" (displayed above the notehead & outside the staff).
2. "LH guitar fingerings" (i.e. left hand, displayed to the left of the notehead).
3. "RH guitar fingerings" (i.e. right hand, displayed in italics above the notehead & outside the staff).
4. "String numbers" (displayed circled above the notehead & outside the staff).
5. Any other text style that the user manually chooses in **Properties > Show more** (display varies according to the chosen style).

![Image](https://github.com/user-attachments/assets/355b0bb7-60d2-485a-bff0-b489bd558f49)

We should check for these styles and customize screen reader output as follows:

1. "Fingering 1"
2. "Fingering 2, left hand guitar"
3. "Fingering 3, right hand guitar"
4. "String number 4"
5. "Fingering 5, [name of text style] style"

For an example, see how the [MusicXML export code differentiates between styles](https://github.com/musescore/MuseScore/blob/a5882b043004d05f884f26b114f7ff05902c8609/src/importexport/musicxml/internal/musicxml/export/exportmusicxml.cpp#L4016-L4044) in order to export these elements correctly.

---

FWIW, musical text in MuseSore is either:

- **Per note** (e.g. fingerings, string numbers)
- **Per chord** (e.g. lyrics, string tunings)
- **Per segment** (e.g. staff text, tempos, expressions, dynamics, chord symbols, sticking, Nashville Number)
- **Per measure** (e.g. rehearsal marks, jumps like "To Coda", "Fine")

We seem to use "Fingering" as the catch-all base element for any kind of per-note text.

**Comment by jmwmusic** [comment]
@shoogle Circled finger numbers are used quite extensively in all junior piano music, especially songs from method books or their supplementary materials, ie those from the Piano Adventures series, Bastien, probably Hal Leonard, and many more.  It signifies a finger number that the hand has to move to a new position in order to play.  I've attached an example from Show Time Popular the Level 2A Piano Adventures popular supplementary book.

Users uploading scores currently just use the string numbers because they look the same.  They open the fingering palette and there is a finger number that looks the same as the one on their score so why wouldn't you!

I was planning on opening a new ticket to request a dedicated circled finger number in the fingering palette that announces with screen reader as "circled finger number".  I could follow up with Sao Mai about correct Braille character once it is there.  Should i still do this separately?

[Happy Birthday - Show Time Popular.zip](https://github.com/user-attachments/files/20073683/Happy.Birthday.-.Show.Time.Popular.zip)

**Comment by shoogle** [comment]
@jmwmusic, thanks! So a circled finger number means:

- In guitar music: a string number.
- In piano music: an ordinary fingering that requires the player to move their hand in order to reach it.

We should probably add a new text style specifically for the piano-type circled fingerings. This way users can customize the two styles independently (e.g. to remove the circles from piano fingerings without removing them from guitar string numbers).

We could call the new style "Reposition fingerings" or "Moved hand fingerings" (feel free to suggest alternative names). Elements with this style would be shown exclusively in the **Fingerings** palette (not in the **Guitar** palette). Within the Fingerings palette, it would go after the plain fingerings but before all guitar-related fingerings.

---

Instead of creating a new style, another option would simply be to interprete elements with the existing "String numbers" style differently depending on whether they are used by a piano instrument or a guitar. This has the advantage that there's only one style with the circled number appearance, so users can't pick the wrong style accidentally. However. it would prevent users from being able to change the appearance of circled piano fingerings independently of guitar string numbers. I think adding a new style is preferable for this reason.

**Comment by jmwmusic** [comment]
Either of those options could work, assuming that if it was interpretted differently for piano than guitar the xml output would pick up this difference for different Braile transcription and different speech output on the score.  This would have the advantage of ensuring people don't pick the wrong one or we might have a number of blind guitarists annoyed when they get weird symbols instead of string numbers if people just choose the first one that is circled.  I have attached the print score original in case there is any obvious difference in how these look that might help.  I don't suppose doing it this way would in any way historically fix existing scores?  If it did that would be an even bigger advantage.

As for what to call them, the junior books would always refer to it as a move or change of hand position.  sometimes text either side of it might say "Move finger 2 to C".  So simply Move Position finger numbers might work well.  once past about Grade 2 pianists work out for themselves whether a move or stretch is required to put that finger on that note.

[Show Time Popular.zip](https://github.com/user-attachments/files/20076205/Show.Time.Popular.zip)
